File: jp-minimal-android/app/src/main/java/br/com/jpminimal/util/Money.kt
package br.com.jpminimal.util

import java.text.NumberFormat
import java.util.Locale

private val BRL: NumberFormat = NumberFormat.getCurrencyInstance(Locale.forLanguageTag("pt-BR"))

fun formatBRL(value: Double): String = BRL.format(value)

const val FREE_SHIPPING_THRESHOLD = 299.0
const val SHIPPING_FIXED = 24.90

fun shippingFor(subtotal: Double, pickup: Boolean): Double {
    if (pickup || subtotal >= FREE_SHIPPING_THRESHOLD) return 0.0
    return SHIPPING_FIXED
}

fun paymentDiscount(subtotal: Double, payment: String, pickup: Boolean): Double {
    if (pickup) return 0.0
    return if (payment == "pix" || payment == "boleto") {
        kotlin.math.round(subtotal * 0.05 * 100.0) / 100.0
    } else {
        0.0
    }
}

File: jp-minimal-ios/JPMinimal/Util/Money.swift
import Foundation

enum Money {
    static let freeShippingThreshold = 299.0
    static let shippingFixed = 24.90

    private static let formatter: NumberFormatter = {
        let f = NumberFormatter()
        f.numberStyle = .currency
        f.locale = Locale(identifier: "pt_BR")
        f.currencyCode = "BRL"
        return f
    }()

    static func formatBRL(_ value: Double) -> String {
        formatter.string(from: NSNumber(value: value)) ?? "R$ 0,00"
    }

    static func shippingFor(subtotal: Double, pickup: Bool) -> Double {
        if pickup || subtotal >= freeShippingThreshold { return 0 }
        return shippingFixed
    }

    static func paymentDiscount(subtotal: Double, payment: String, pickup: Bool) -> Double {
        if pickup { return 0 }
        if payment == "pix" || payment == "boleto" {
            return (subtotal * 0.05 * 100).rounded() / 100
        }
        return 0
    }
}
